### ISSUE TITLE
NPE in completion contributor

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/sdk#25331]
It appears that the export namespace is not always computed for all libraries when we get to this method.

Dart analysis server, SDK version 1.14.0-edge.6602644bb3fd094cddc638a8e963c6a49d9cbdc8, server version 1.14.0, error: Failed to handle completion domain request: {clientRequestTime: 1451694275093, params: {file: .../scope.dart, offset: 2622}, method: completion.getSuggestions, id: 277}
The null object does not have a getter 'definedNames'.

NoSuchMethodError: method not found: 'definedNames'
Receiver: null
Arguments: []
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
#2      ImportedReferenceContributor._buildSuggestions (package:analysis_server/src/services/completion/dart/imported_reference_contributor.dart:67)
#3      ImportedReferenceContributor.computeSuggestions.<computeSuggestions_async_body> (package:analysis_server/src/services/completion/dart/imported_reference_contributor.dart:52)
#4      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:27)
#5      _rootRunUnary (dart:async/zone.dart:914)
#6      _CustomZone.runUnary (dart:async/zone.dart:810)
#7      _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:551)
#8      _Future._propagateToListeners (dart:async/future_impl.dart:637)
#9      _Future._complete (dart:async/future_impl.dart:414)
#10     _SyncCompleter.complete (dart:async/future_impl.dart:52)
#11     DartCompletionRequestImpl.resolveDirectives.<resolveDirectives_async_body> (package:analysis_server/src/services/completion/dart/completion_manager.dart)
#12     Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
#13     _rootRun (dart:async/zone.dart:903)
#14     _CustomZone.run (dart:async/zone.dart:802)
#15     _CustomZone.runGuarded (dart:async/zone.dart:708)
#16     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:733)
#17     _rootRun (dart:async/zone.dart:907)
#18     _CustomZone.run (dart:async/zone.dart:802)
#19     _CustomZone.runGuarded (dart:async/zone.dart:708)
#20     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:733)
#21     _microtaskLoop (dart:async/schedule_microtask.dart:41)
#22     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#23     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#24     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)

**Comment by alexander-doroshko** [comment]
@danrubel should this be fixed in 1.14.2? I'm asking because we've got similar trace from user:
```
Dart analysis server, SDK version 1.14.2, server version 1.14.0, error: Failed to handle completion domain request: {clientRequestTime: 1456183637438, params: {file: C:\Users\Adrian\Desktop\nnchat44\src\main\resources\web\main.dart, offset: 3543}, method: completion.getSuggestions, id: 437}
The null object does not have a getter 'definedNames'.

NoSuchMethodError: method not found: 'definedNames'
Receiver: null
Arguments: []
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
#2      ImportedReferenceContributor._buildSuggestions (package:analysis_server/src/services/completion/dart/imported_reference_contributor.dart:67)
#3      ImportedReferenceContributor.computeSuggestions.<computeSuggestions_async_body> (package:analysis_server/src/services/completion/dart/imported_reference_contributor.dart:52)
#4      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:27)
#5      _rootRunUnary (dart:async/zone.dart:914)
#6      _CustomZone.runUnary (dart:async/zone.dart:810)
#7      _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:551)
#8      _Future._propagateToListeners (dart:async/future_impl.dart:637)
#9      _Future._complete (dart:async/future_impl.dart:414)
#10     _SyncCompleter.complete (dart:async/future_impl.dart:52)
#11     DartCompletionRequestImpl.resolveDirectives.<resolveDirectives_async_body> (package:analysis_server/src/services/completion/dart/completion_manager.dart:218)
#12     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:27)
#13     _rootRunUnary (dart:async/zone.dart:914)
#14     _CustomZone.runUnary (dart:async/zone.dart:810)
#15     _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:551)
#16     _Future._propagateToListeners (dart:async/future_impl.dart:637)
#17     _Future._completeWithValue (dart:async/future_impl.dart:424)
#18     _Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:479)
#19     _rootRun (dart:async/zone.dart:907)
#20     _CustomZone.run (dart:async/zone.dart:802)
#21     _CustomZone.runGuarded (dart:async/zone.dart:708)
#22     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:733)
#23     _microtaskLoop (dart:async/schedule_microtask.dart:41)
#24     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#25     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#26     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)
```

**Comment by danrubel** [comment]
I don't think the fix made it into 1.14.2. I believe it's fixed in 1.15.0
